infrastructure/src/main/java/org/wso2/testgrid/infrastructure/aws/AWSManager.java

```diff
@@ -249,24 +249,18 @@ public boolean destroyInfrastructure(Script script) throws TestGridInfrastructur
             cfCompatibleParameters.add(awsParam);
         });
 
-        for (Parameter parameter : cfCompatibleParameters) {
-            if (parameter.getParameterKey().equals(WUM_USERNAME)) {
-                String wumUserName = EnvironmentUtil.getSystemVariableValue(parameter.getParameterValue());
-                if (wumUserName != null) {
-                    parameter.setParameterValue(wumUserName);
-                } else {
-                    throw new TestGridInfrastructureException("WUM Credentials (WUM_USERNAME, WUM_PASSWORD) must be "
-                            + "set as environment variables");
-                }
+        script.getEnvironmentScriptParameters().forEach((key, value) -> {
+            String envVariable = EnvironmentUtil.getSystemVariableValue((String) value);
+            if (envVariable != null) {
+                Parameter awsParam = new Parameter().withParameterKey((String) key).withParameterValue(envVariable);
+                cfCompatibleParameters.add(awsParam);
+            } else {
+                log.error("Environment Variable " + value + "not found !!");
```

Comment:
It is better to throw an exception from here. Please refer [1] for throwing exceptions from lambda 

[1] https://github.com/wso2/carbon-utils/blob/master/components/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/LambdaExceptionUtils.java